io-engine/src/lvs/lvs_lvol.rs

```diff
@@ -1026,4 +1042,60 @@ impl LvsLvol for Lvol {
         }
         Ok(name)
     }
+
+    /// Resize a replica. The resize can be expand or shrink, depending
+    /// upon if required size is more or less than current size of
+    /// the replica.
+    async fn resize_replica(mut self, resize_to: u64) -> Result<Lvol, Error> {
+        let (s, r) = pair::<ErrnoResult<*mut spdk_lvol>>();
+        let mut ctx = ResizeCbCtx {
+            lvol: self.as_inner_ptr(),
+            sender: cb_arg(s),
+            req_size: resize_to,
+        };
+
+        unsafe {
+            vbdev_lvol_resize(
+                self.as_inner_ptr(),
+                resize_to,
+                Some(lvol_resize_cb),
+                &mut ctx as *mut _ as *mut c_void,
+            );
+        }
+
+        let cb_ret = r.await.expect("lvol resize callback dropped");
+
+        if let Ok(_lv) = cb_ret {
+            info!("Resized {:?} successfully", self);
+            Ok(Lvol::from_inner_ptr(self.as_inner_ptr()))
+        } else {
+            let status = cb_ret.unwrap_err();
+            error!("Resize {:?} failed, errno {}", self, status);
+            Err(Error::RepResize {
+                source: status,
+                name: self.name(),
+            })
+        }
```

Comment:
Use match instead of if else

```suggestion
        match cb_ret {
            Ok(_) => {
                info!("Resized {self:?} successfully");
                Ok(Lvol::from_inner_ptr(self.as_inner_ptr()))
            }
            Err(errno) => {
                error!("Resize {self:?} failed, errno {errno}");
                Err(Error::RepResize {
                    source: errno,
                    name: self.name(),
                })
            }
        }
```